themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +28,35 @@
     <label><?=$this->translate('Sort') ?></label>
     <div class="btn-group">
       <?php foreach ($this->sortOptions as $key => $sort): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) ?>" class="btn btn-default js-facet-sort<?php if ($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
+        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) . '&amp;contains=' . urlencode($contains) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>"><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
   </div>
 <?php endif; ?>
-<div class="lightbox-scroll full-facets">
+
+<!-- Hopefully we can find a better way for passthrough of these variables -->
```

Comment:
Are all these input elements currently existing outside of a `<form>` tag? Is it worth wrapping them in a form so that there's some potential to make this work in a non-Javascript scenario? (Note that we're not really too concerned about making everything work without Javascript, as that's not very realistic these days... but if it's easy to do, it doesn't hurt and can be useful for double-checking things when there's a problem).

If we don't need a form at all, is there a strong reason to use hidden inputs for the values you want to pass forward? Could this instead potentially be a single hidden element with many `data-` attributes, for example? I'm not sure that that is a huge improvement, but it might at least be somewhat less verbose.

Reply:
Do you mean something like adding a button that somebody would have to click to reload the page which would basically be the replacement of the keyupCallback in a non-JS scenario?

I tried to merge the different hidden inputs together with a schema like this:
`<input class="ajax_param" type="hidden" data-facet="<?=$this->facet?>" data-searchAction="<?=$searchAction?>" ...>`

The problem is: HTML Attributes should always be lowercase. Even when written in camelCase, the jQuery API will lowercase them which will make it harder to generically pass them through via AJAX to PHP again with the right naming. Do you have any more precise suggestion how the HTML should look like?

Reply:
How about encoding all the parameters into a single element as JSON with `HtmlSafeJsonEncode`?

Reply:
Thanks, i didn't even know that helper existed! Now i changed it accordingly.

However, the question regarding a non-JS scenario still seems to be open. But do we really need to support that? Are there any other examples in the code for similar solutions?

Reply:
@mtrojan-ub, I don't think there's a strong need to support the non-JS scenario -- it's something I like to do when it's easy (since generally, a non-JS-friendly solution tends to mean less JS to maintain overall), but sometimes it's not easy and not worth the hassle. My original question was motivated less by concern for offering non-JS support here and more out of concern about whether free-floating input elements might pose some kind of accessibility or HTML validation problem. (I'm not saying they will -- just wondering if the semantic container of a form around related inputs potentially offers any benefits here).

Reply:
I'm not sure either way will provide any disadvantage (e.g. for screen readers), but since it at least improves the readability of the code a little bit, i guess it's OK to add it.